pkg/cmd/config/config.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2024 The KitOps Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"kitops/pkg/output"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+)
+
+type Config struct {
+	LogLevel  string `json:"log_level"`
+	Progress  string `json:"progress"`
+	ConfigDir string `json:"config_dir"`
+}
+
+// DefaultConfig returns a Config struct with default values.
+func DefaultConfig() *Config {
+	return &Config{
+		LogLevel:  output.LogLevelInfo.String(),
+		Progress:  "plain",
+		ConfigDir: "",
+	}
+}
+
+// Set a configuration key and value.
+func setConfig(_ context.Context, opts *configOptions) error {
+	configPath := getConfigPath(opts.profile)
+	cfg, err := LoadConfig(configPath)
+	if err != nil {
+		cfg = DefaultConfig() // Start with defaults if config doesn't exist.
+	}
```

Comment:
We generally don't want to save defaults to the file and instead use them internally only. Saving defaults has a few downsides: 

1. We have no way of knowing which fields are actually user-specified and which are just there because they're an option
2. It makes the file more bloated and harder to understand (imagine there's a dozen configurable fields, for example)
3. We can't update any default values, since any user who has run the config command will have the old defaults saved to a file, and we have no way of knowing which files are safe to update

Reply:
Refactored to only use the default values internally instead of saving them to a config file